activerecord/lib/active_record/backtrace_cleaner.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "active_support/backtrace_cleaner"
+
+module ActiveRecord
+  class BacktraceCleaner < ActiveSupport::BacktraceCleaner
```

Comment:
Why we have this one give we only use the `Rails.backtrace_cleaner`?

Reply:
@rafaelfranca because of https://github.com/rails/rails/pull/26815#discussion_r84569506 :wink:

Reply:
In that comment was about direct access to the rails constant inside the log subscriber, but if that access was in the Railtie as it is in this patch it is fine. This cleaner is not used in this patch, only in the tests. I think it is better to use the `Rails::BacktraceCleaner` in the tests and have the default value in Active Record to be an instance of `ActiveSupport::BacktraceCleaner`

Reply:
@rafaelfranca The `ActiveSupport::BacktraceCleaner` doesn't do any filtering or silencing be default so if it was to be used as is, anyone using the verbose query logging outside of Rails would see `lib/active_record/log_subscriber.rb:98` as the caller for every query. I don't think that is it a good developer experience to have to configure the backtrace cleaner to get anything meaningful out of these logs.

It would be possible to use a `ActiveSupport::BacktraceCleaner` and manually configure it within the LogSubscriber to ignore the paths, but I'm not sure what the advantage is over the current approach.

Alternatively, we could enhance the `ActiveSupport::BacktraceCleaner` so that it can be used unmodified. For this, I would suggest:

* Move the concept of `@root` to `ActiveSupport::BacktraceCleaner`. It can default to working directory. `Rails::BacktraceCleaner` can overwrite it to be `Rails.root`.
* Change the './' => '/' to be './' => '' so that it is clear that things in `./` are in the working directory. (I think that this is why the current `APP_DIRS_PATTERN` allows starting with `/`, which I don't think that it should).
* Replace `APP_DIRS_PATTERN` silencer with one that rejects anything beginning with `/` (is outside of the app), `vendor/` (this is not our code) or `bin` (these get generated by gems).
* `add_gem_filters` can be moved to `ActiveSupport::BacktraceCleaner` as it is still useful if working outside of rails).

This leaves `Rails::BacktraceCleaner` with setting the root to Rails.root and cleaning out the ActionView template methods.

It would still be possible to get a noop backtrace cleaner using `remove_silencers!` and `remove_filters!`, so it would still have all the capabilities, just that the default stack would be more useful. 

Let me know if you think either of these approaches sounds ok.

Reply:
I'm a little reluctant to invent the concept of an application root outside of Rails-proper, but I think gem/stdlib filtering is compelling functionality for AS::BacktraceCleaner... and that alone should be enough for us here?